### PR TITLE
feat: add awakening mask progress bar

### PR DESCRIPTION
**Pull request by byBackfish**
Ingame view (left is blood bar, right is awakening bar):
image

Mask of Awakening description:
 image

### CODE REVIEW

**Comment by byBackfish** [comment]
Oh, I always read it as `mask of awakening` instead of `mask of awakened`, I'll fix this "typo" later today

File: src/main/java/com/wynntils/modules/utilities/configs/OverlayConfig.java [hunk 1]
```diff
@@ -172,6 +172,46 @@ public enum BloodPoolTextures {
...
+        @Setting.Limitations.IntLimit(min = 0, max = 81)
+        public int width = 81;
+
+        @Setting(displayName = "Awakening Progress Orientation", description = "How orientated in degrees should the awakening progress bar be?\n\n§8Accompanied text will be removed.")
+        public OverlayRotation overlayRotation = OverlayRotation.NORMAL;
+
+        @Setting(displayName = "Awakening Progress Texture", description = "What texture should be used for the awakening progress bar?")
+        public AwakenedProgressTexture awakenedProgressTexture = AwakenedProgressTexture.a;
+
+        @Setting.Limitations.FloatLimit(min = 0f, max = 10f)
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/configs/OverlayConfig.java]: Don't we usually put this below \@Setting? Not that it matters though

File: src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    public Pair<Integer, Integer> textPositionOffset = new Pair<>(-40, -10);
+
+    @Setting(displayName = "Text Name", description = "What should the colour of the text be?")
+    public CustomColor textColor = CommonColors.WHITE;
+
+    private static float awakeningProgress = 0.0f;
+
+    @Override
+    public void tick(TickEvent.ClientTickEvent event, long ticks) {
+        if (!(visible = (get(CharacterData.class).getCurrentAwakenedProgress() != -1 && !Reference.onLobby))) return;
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java:40]: Why visible = ?

File: src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                break;
+            case Rune:
+                drawDefaultBar(-1, 8, 148, 163, textColor);
+                break;
+        }
+    }
+
+    private void drawDefaultBar(int y1, int y2, int ty1, int ty2, CustomColor cc) {
+        if (OverlayConfig.AwakenedProgress.INSTANCE.overlayRotation == OverlayRotation.NORMAL) {
+            drawString(get(CharacterData.class).getCurrentAwakenedProgress() + " ۞ " + get(CharacterData.class).getMaxAwakenedProgress(), textPositionOffset.a  - (81- OverlayConfig.AwakenedProgress.INSTANCE.width), textPositionOffset.b, cc, SmartFontRenderer.TextAlignment.MIDDLE, OverlayConfig.AwakenedProgress.INSTANCE.textShadow);
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java]: 81 -

**Comment by kristofbolyai** [comment]
Please use conventional name for the PR too (chore,fix,feat,docs).

File: src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java (hunk 2 above)
Comment (byBackfish) [reply src/main/java/com/wynntils/modules/utilities/overlays/hud/AwakenedProgressBarOverlay.java:40]: Same is done in [`BloodPoolBarOverlay.java`](https://github.com/Wynntils/Wynntils/blob/1ea89a396862268ad2aa342fea3a90a12a8d1cff/src/main/java/com/wynntils/modules/utilities/overlays/hud/BloodPoolBarOverlay.java#L40), i just  copied the general code over from there

**Comment by kristofbolyai** [comment]
I don't know what the boss bar looks like, but white text is weird for a red bar. But at the same time, we have 3 red bars now..

File: src/main/java/com/wynntils/modules/core/events/ClientEvents.java [hunk 4]
```diff
@@ -329,13 +328,21 @@ public void updateActionBar(PacketEvent<SPacketChat> e) {
     public void updateBossBar(PacketEvent<SPacketUpdateBossInfo> e) {
         if (!Reference.onServer) return;
 
-        PlayerInfo.get(BossBarData.class).updateBloodPoolBar(e.getPacket());
+        PlayerInfo.get(BossBarData.class).updateBossbarStats(e.getPacket());
 
-        if (OverlayConfig.BloodPool.INSTANCE.hideDefaultBar && e.getPacket() != null && e.getPacket().getName() != null) {
+        if (e.getPacket() == null ||e.getPacket().getName() == null) return;
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/core/events/ClientEvents.java]:
```suggestion
        if (e.getPacket() == null || e.getPacket().getName() == null) return;
```

**Review by kristofbolyai — APPROVED** [review]

**Review by kristofbolyai — APPROVED** [review]

**Review by HighCrit — APPROVED** [review]

**Pull request merged**